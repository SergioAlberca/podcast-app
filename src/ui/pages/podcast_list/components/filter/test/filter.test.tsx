import { vi } from "vitest";
import { fireEvent, render, screen } from "@/test/test_utils";
import Filter from "../filter";

describe("Filter component test", () => {
  it("render component", () => {
    const { container } = render(<Filter payload="fake" setPayload={() => "fake"} podcastLength={0} />);
    expect(container).toMatchSnapshot();
  });

  it("Payload to be in the document", () => {
    render(<Filter payload="fake" setPayload={() => "fake"} podcastLength={0} />);
    expect(screen.getByDisplayValue("fake")).toBeInTheDocument();
  });

  it("SetPayload has been called", () => {
    const setValue = vi.fn();
    const { getByTestId } = render(<Filter payload="" setPayload={() => setValue("")} podcastLength={0} />);
    const input = getByTestId("filter-input") as HTMLInputElement;
    fireEvent.change(input, { target: { value: setValue("fakecustom") } });
    expect(setValue).toHaveBeenCalled();
  });
});
