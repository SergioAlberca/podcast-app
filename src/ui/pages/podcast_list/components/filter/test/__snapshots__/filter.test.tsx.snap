// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Filter component test > render component 1`] = `
<div>
  <div
    class="_input-container_da77d8"
  >
    <span />
    <input
      class="_input-text_da77d8"
      data-testid="filter-input"
      id="input"
      placeholder="Filter podcasts..."
      type="text"
      value="fake"
    />
  </div>
</div>
`;
